Essentials/src/com/earth2me/essentials/utils/LocationUtil.java

```diff
@@ -17,8 +17,15 @@
 
 public class LocationUtil {
     // Water types used for TRANSPARENT_MATERIALS and is-water-safe config option
-    private static final Set<Material> WATER_TYPES =
-            EnumUtil.getAllMatching(Material.class, "WATER", "FLOWING_WATER");
+    private static final Set<Material> WATER_TYPES = EnumUtil.getAllMatching(Material.class,
+            "WATER", "FLOWING_WATER");
+
+    // Types checked by isBlockDamaging
+    private static final Set<Material> DAMAGING_TYPES = EnumUtil.getAllMatching(Material.class,
+            "CACTUS", "MAGMA_BLOCK", "CAMPFIRE", "SOUL_CAMPFIRE", "FIRE", "SOUL_FIRE", "SWEET_BERRY_BUSH");
+    private static final Set<Material> LAVA_TYPES = EnumUtil.getAllMatching(Material.class,
+            "LAVA", "FLOWING_LAVA", "STATIONARY_LAVA");
+    private static final Material PORTAL = EnumUtil.getMaterial("NETHER_PORTAL", "PORTAL");
```

Comment:
above and end portals?

Reply:
I think this is intended only to match a Nether portal. In `isBlockDamaging` it's checked, presumably so that players are not teleported into a block of a portal, where you can essentially get stuck and not be able to type commands or break blocks anymore.